Convert mappings to SSSOM files

cthoyt opened the pull request:
This PR adds a script that converted the original mapping CSV files into standardized, self-contained SSSOM (Simple Standard for Sharing Ontological Mappings).

There are still a few TODOs/questions to finish this:

- [x] what's the USDA namespace? We can register this in the bioregistry too. (https://github.com/biopragmatics/bioregistry/pull/1562)
- [ ] How were these mappings produced? SSSOM can capture more information about this
- [x] Add mapping set IDs, titles, and other SSSOM-metadata (done in 54943a4)
- [ ] What is going on in the RxNorm extensions file? The IDs have OMOP in their prefixes.

cthoyt committed:
Add SSSOM conversion

zhengj2007 commented:
@cthoyt  Thanks for making the script and generating the SSSOM files.

Here are my answers for TODOs you listed. 
 
 * [ ]  what's the USDA namespace? We can register this in the bioregistry too.

We found the USDA licensed veterinary vaccines from the USDA catalog (https://www.aphis.usda.gov/veterinary-biologics/licensed-products). Each vaccine has an ID with prefix USDA, such as USDA: 1431.50. I think we can use 'USDA' (which stand for U.S. Department of Agriculture) as namespace .

 * [ ]  How were these mappings produced? SSSOM can capture more information about this

We performed automated mapping but only few mapped terms were identified. (I will find this list.) Most of these vaccines from different resources were added into the VO by aligning with VO hierarchy. I have no idea what I should fill in this field.

 * [ ]  Add mapping set IDs, titles, and other SSSOM-metadata

Could you please give me some examples for title fields?

 * [ ]  What is going on in the RxNorm extensions file? The IDs have OMOP in their prefixes.

RxNorm extensions were developed by OHDSI. The ID with OMOP prefix OMOP seems assigned by OHDSI. The OMOP Concept IDs are identifiers assigned to concepts within the OMOP Common Data Model.

Could you please give me more instructions for finishing the TODO list? Thanks!

cthoyt committed:
Update mapping generation

cthoyt committed:
Update USDA prefix based on https://github.com/biopragmatics/bioregistry/pull/1562

cthoyt commented:
Thanks for the update on the USDA. I added a new prefix to the Bioregistry to support this in https://github.com/biopragmatics/bioregistry/pull/1562 and updated the SSSOM generation script in dfa666e.

So to clarify, these mappings were ultimately manually curated? Were they all done by you @zhengj2007?

Follow-up question about the RxNorm extensions file - if these are OMOP identifiers, how is it different from the OMOP mapping file?

cthoyt committed:
Pass sssom validate

zhengj2007 commented:
> Thanks for the update on the USDA. I added a new prefix to the Bioregistry to support this in [biopragmatics/bioregistry#1562](https://github.com/biopragmatics/bioregistry/pull/1562) and updated the SSSOM generation script in [dfa666e](https://github.com/vaccineontology/VO/commit/dfa666e213fd5aad0ec1d3a41cb1d28da6e4b1c0).

Thanks for adding the USDA namespace.

> 
> So to clarify, these mappings were ultimately manually curated? Were they all done by you @zhengj2007?

FDA, CVX code set, and USDA are manually curated. For RxNorm and RxNorm extension, we used script to extract some vaccine information from the term labels and added the terms in a template for ROBOT conversion. It more likes semi-automatic approach.

I am not the only person worked on the mappings. The term editors indicate the persons who did the mappings. I don't know whether I can collect the ORCID IDs for all the persons contributed on the mappings. Besides, the term editors for most OMOP terms are OHDSI vaccine WG (working group). Do we need an ORCID for it? 

In addition, It's hard to find out when the mappings were performed. Recently, I updated few mappings since the previous mappings are not correct. 

> Follow-up question about the RxNorm extensions file - if these are OMOP identifiers, how is it different from the OMOP mapping file?

It seems both IDs are used in OHDSI database. I will contact OHDSI people for this question and let you know the answer asap.

Thanks again for your help with it @cthoyt

zhengj2007 commented:
> @zhengj2007 any updates?

@cthoyt please check my previous comments. Any suggestions on what ORCID should be used for a work group? 

I am going to fill up the information as much as I can soon. Will let you know when I am done.

zhengj2007 commented:
SSSOM files have been added
